### DISCUSSION TITLE
Newly setup turborepo doesn't see packages with npm and yarn

### DISCUSSION THREAD

**Discussion by idbd1** [vercel/turborepo#5869] [Help] [answered]
when I run any turbo command from the root using npm (yarn also doesn't work), the result is
```
➜  headcount git:(setup-turborepo) ✗ turbo build
• Packages in scope: 
• Running build in 0 packages
• Remote caching disabled

No tasks were executed as part of this run.

 Tasks:    0 successful, 0 total
Cached:    0 cached, 0 total
  Time:    40ms 
```

in the root, package.json looks like that:
```
{
  "name": "root",
  "private": true,
  "scripts": {
    "build": "turbo run build",
    "start": "turbo run start",
    "lint": "turbo run lint",
    "format": "prettier --write \"**/*.{ts,tsx,md}\""
  },
  "devDependencies": {
    "eslint-config-custom": "*",
    "prettier": "3.0.3",
    "turbo": "latest"
  },
  "workspaces": ["ui/*"]
}
```
turbo.json in the root
```
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**"]
    },
    "lint": {}
  }
}
```

in the ui folder package.json 
```
"name": "ui",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "build": "vite build",
    "start": "vite",
    "lint": "eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "ts": "tsc",
    "preview": "vite preview"
   "dependencies": { ... }
  },
```

I can't figure out what's the problem, no matter the command executed from the root, it doesn't seem to work. I tried changing turbo version in the root to v. 1.10.4, 1.9.1, (I rm -rf node_modules, installed in the root), but it didn't help.

**Comment by mehulkar** [comment, marked as answer]
The key pieces here are:

- `"workspaces": ["ui/*"]` in package.json
- where your workspace is. 

You mention:

> in the ui folder package.json

If that means you have `ui/package.json`, that is likely the problem. You can change your `workspaces` config to just `["ui"]` to solve this. `ui/*` means that you have multiple directories inside `ui/` and each of them are a workspace. 

See the npm docs on workspaces here: https://docs.npmjs.com/cli/v10/using-npm/workspaces#defining-workspaces
